src/structures/gdt.rs

```diff
@@ -105,6 +105,38 @@ impl GlobalDescriptorTable {
         }
     }
 
+    /// Forms a GDT from a slice of `u64`.
+    ///
+    /// # Safety
+    ///
+    /// * The user must make sure that the entries are well formed
+    /// * The provided slice **must not be larger than 8 items** (only up to the first 8 will be observed.)
+    #[inline]
+    #[cfg(feature = "nightly")]
+    pub const unsafe fn from_raw_parts(slice: &[u64]) -> GlobalDescriptorTable {
+        assert!(
+            slice.len() <= 8,
+            "initializing a GDT from a slice requires it to be **at most** 8 elements."
+        );
+        let next_free = slice.len();
+
+        let mut table = [0; 8];
+        let mut idx = 0;
+
+        while idx != next_free {
+            table[idx] = slice[idx];
+            idx += 1;
+        }
+
+        GlobalDescriptorTable { table, next_free }
+    }
+
+    /// Get a reference to the internal table.
```

Comment:
Could you add a note to the docs that the resulting slice might contain system descriptors, which span two `u64`s? We want to avoid that users think that the length of the slice is equal to the number of GDT entries.